### ISSUE TITLE
Not compatible with ES2015 / ES6 (import)

### ISSUE THREAD

**Issue by sam-s4s** [Axinom/x2js#61]
Hmm, I would be great if you could use this package in ES2015 / ES6 land (ie. with Rollup).

Currently only seems to be able to be used with require(), which isn't compatible.

Needs to be able to be imported...

import x2js from 'x2js';
-or even-
import * as x2js from 'x2js';

Any chance of that happening? :)

Also typings? :D

**Comment by Maxouhell** [comment]
It works for me, do you have a repo where we can check ?
What version do you use ?

**Comment by sam-s4s** [comment]
Oh it looks like the new version has typings and stuff :D

I'll have to try it out! Sounds promising!

**Comment by sam-s4s** [comment]
Ok so I'm now doing this...
```
import * as X2JS from 'x2js';
...
let xml = '<root><dog>7</dog></root>';
let x2js = new X2JS();
console.log(x2js.xml2js(xml));
```

And when my app loads it gives the following error (while initially loading in the X2JS code - not running my code).

Uncaught TypeError: Cannot set property 'X2JS' of undefined

from this line...

root.X2JS = factory();

**Comment by Maxouhell** [comment]
Hum... My Angular4 project is on webpack2 / Rollup and it works with both.
I don't thing I have something special to make it work.

Are you on x2js@3.1.1 ?

I can make a little example project if you want.

**Comment by sam-s4s** [comment]
Ah interesting. Yeah, I just tested with 3.1.1.

I wonder what the difference is??

That would be great, thanks :D

**Comment by RangerRick** [comment]
I get the same error using TypeScript + JEST for testing, but it passes in my actual webpack 2 build.

**Comment by Maxouhell** [comment]
Hi !
Sorry, i'm late.

So this is a test with rollup, webpack and X2JS : 
https://github.com/Maxouhell/x2js-rollup-webpack-test

**Comment by sam-s4s** [comment]
Hmm, the end result of that build is very different...

I'm pushing out a single bundled js file - whereas you're referencing everything directly from the node_modules directory, etc. (kind of avoiding all the problems we're having, however obviously this is completely inappropriate for a production build).

Unless I'm mistaken? :P

For example, you're not actually compiling or bundling either angular or x2js at all. So you won't have any problems, because you're not actually doing anything ^_^

Should I post some code of what I'm doing?

**Comment by Maxouhell** [comment]
Yes please, I just took a Yeoman to generate the Rollup project, and I didn't update the config.
If you can give me yours, I will try with it :)

**Comment by sam-s4s** [comment]
Hey @Maxouhell :)

I have uploaded a test for you...

https://github.com/sam-s4s/AngularTest

Should be a good example to work with! I tried to make it as similar to my actual production build as possible :)

Thanks!
